tools/build_api.py

```diff
@@ -283,7 +291,7 @@ def get_mbed_official_release(version):
                 [
                     TARGET_MAP[target].name,
                     tuple(transform_release_toolchains(
-                        TARGET_MAP[target].supported_toolchains, version))
+                        TARGET_MAP[target].supported_toolchains, version, target))
```

Comment:
I'm pretty sure that this is passing in the target string, and not the target object. Since `transform_release_toolchains` is only ever called here, how about changing the function to take `target, version` instead of `supported_toolchain_list, version, target`? That way we don't have to do lookup 3(!) times.

Reply:
Is there a reason for not using ````TARGET_MAP[target].supported_toolchains````, as in https://github.com/ARMmbed/mbed-os/pull/9878?
Does targets.json only define mbed2 supported toolchains?

Reply:
@alekla01 - Please see my changes in this file to take care of previous tools version as well. 
But you asked great question on why we are not using 
supported toolchains. I would like to investigate why we are not just return 
supported toolchain. The current attempt is to keep the behavior same as before as it might impact online compiler.